src/arrayvec.rs

```diff
@@ -1502,3 +1541,39 @@ impl<A: Array> ArrayVec<A> {
     self.drain_to_vec_and_reserve(0)
   }
 }
+
+#[cfg(feature = "serde")]
+struct ArrayVecVisitor<A: Array>(PhantomData<A>);
+
+#[cfg(feature = "serde")]
+impl<'de, A: Array> Visitor<'de> for ArrayVecVisitor<A>
+where
+  A: Default,
+  A::Item: Deserialize<'de>,
+{
+  type Value = ArrayVec<A>;
+
+  fn expecting(
+    &self,
+    formatter: &mut core::fmt::Formatter,
+  ) -> core::fmt::Result {
+    formatter.write_str("a sequence")
+  }
+
+  fn visit_seq<S>(self, mut seq: S) -> Result<Self::Value, S::Error>
+  where
+    S: SeqAccess<'de>,
+  {
+    let mut new_arrayvec: ArrayVec<A> = Default::default();
+
+    while let Some(value) = seq.next_element()? {
+      // Prefer to truncate rather than panic.
```

Comment:
`arrayvec::ArrayVec` returns an "unknown error" in this case. I have no idea how the serde API lets you do that, but seems preferably to silent truncation.

Reply:
Ah, with `std` active it gives an error message:

```
invalid length 5, expected an array with no more than 4 items at line 1 column 15
```

Reply:
Ah, the error trait has constructors: https://docs.rs/serde/1.0.115/serde/de/trait.Error.html#method.invalid_length